### ISSUE TITLE
"ResourceWarning: unclosed file" when using Graph.parse() with python3

### ISSUE THREAD

**Issue by cmaumet** [RDFLib/rdflib#651]
Hi,

With python 3, when invoking the `parse()` function on turtle files as part of our testing suite, we get the following warning:
```
ResourceWarning: unclosed file <_io.BufferedReader name='<PATH_TO_TTL_FILE>'>
``` 
The full test report is available [here](https://travis-ci.org/incf-nidash/nidmresults-fsl/builds/157843978). 

The warning can be reproduced with this smaller piece of code:
```
$ python test_rdflib_warning.py 
.test_rdflib_warning.py:23: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/file.ttl'>
  g2.parse(local_ttl_file, format='turtle')
.
----------------------------------------------------------------------
Ran 2 tests in 3.709s

OK
```

Where `test_rdflib_warning.py ` contains:
```
from rdflib.graph import Graph
import urllib.request
import unittest


class TestResourceWarning(unittest.TestCase):

    def test_url(self):
        # No warning
        ttl_url = \
            "https://provenance.ecs.soton.ac.uk/store/documents/114154.ttl"

        g1 = Graph()
        g1.parse(ttl_url, format='turtle')

    def test_url_local_copy(self):
        # Warning
        # test_rdflib_warning.py:23: ResourceWarning: unclosed file
        # <_io.BufferedReader name='/tmp/file.ttl'>
        ttl_url = \
            "https://provenance.ecs.soton.ac.uk/store/documents/114154.ttl"

        local_ttl_file = '/tmp/file.ttl'
        urllib.request.urlretrieve(ttl_url, '/tmp/file.ttl')

        g2 = Graph()
        g2.parse(local_ttl_file, format='turtle')

if __name__ == '__main__':
    unittest.main()
```
We are using rdflib 4.2.1 (latest version on pypi).

Is it possible that the turtle parser is not closing the turtle file after reading it?

**Comment by cmaumet** [comment]
(In case this is of interest the tests were run as part of to https://github.com/incf-nidash/nidmresults-fsl/pull/93).

**Comment by joernhees** [comment]
can't reproduce with current dev version:
```
$ python
Python 3.5.2 (default, Jun 29 2016, 13:43:58)
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from rdflib.graph import Graph
INFO:rdflib:RDFLib Version: 4.2.2-dev
>>> import urllib.request
>>> ttl_url="https://provenance.ecs.soton.ac.uk/store/documents/114154.ttl"
>>> g1 = Graph()
>>> g1.parse(ttl_url, format='turtle')
<Graph identifier=N1dddfc5c48794ecaa5462852a135e9ec (<class 'rdflib.graph.Graph'>)>
>>> local_ttl_file = '/tmp/file.ttl'
>>> urllib.request.urlretrieve(ttl_url, local_ttl_file)
('/tmp/file.ttl', <http.client.HTTPMessage object at 0x1076d6d68>)
>>> g2 = Graph()
>>> g2.parse(local_ttl_file, format='turtle')
<Graph identifier=Ne80a22bbb8b34ca9bd2e119c0c2983ed (<class 'rdflib.graph.Graph'>)>
>>>
```

i guess this was closed in #518 already
